lib/ddtrace/transport/http/adapters/ethon.rb

```diff
@@ -0,0 +1,98 @@
+require 'ddtrace/transport/response'
+require 'ethon'
+
+module Datadog
+  module Transport
+    module HTTP
+      module Adapters
+        # Adapter for Ethon >= 0.8.0.
+        # More memory efficient than Net::HTTP.
+        class Ethon
+          attr_reader \
+            :hostname,
+            :port,
+            :timeout
+
+          DEFAULT_TIMEOUT = 10
+
+          def initialize(
+            hostname = Datadog::Transport::HTTP.default_hostname,
+            port = Datadog::Transport::HTTP.default_port,
+            options = {}
+          )
+            @hostname = hostname
+            @port = port
+            @timeout = options[:timeout] || DEFAULT_TIMEOUT
+
+            @uri = URI::HTTP.build(host: hostname, port: port).to_s
+          end
+
+          def call(env)
+            if respond_to?(env.verb)
+              send(env.verb, env)
+            else
+              raise UnknownHTTPMethod, env
+            end
+          end
+
+          def post(env)
+            client = ::Ethon::Easy.new
```

Comment:
According to https://github.com/typhoeus/typhoeus/blob/a345545e5e4ac0522b883fe0cf19e5e2e807b4b0/lib/typhoeus/pool.rb#L5 , creation of `Easy` instances is expensive.

Typhoeus seems to implement a whole pool and synchronization around reuse.

I guess in our case the question is: how often are these http requests done? Would it be worth to reuse these instances (either by doing it manually or switching to typhoeus)?

Reply:
I think the requests right now are done every second.

Reply:
Thanks for clarifying. I guess in that case it may be ok-ish to just create a new instance every time. Perhaps just leaving a comment in the code for the future documenting that this may become a performance concern :)